provider/core/src/error.rs

```diff
@@ -230,19 +235,6 @@ impl DataError {
         self
     }
 
-    /// Logs the data error with the given context, then return self.
-    ///
-    /// This does not modify the error, but if the "log_error_context" feature is enabled,
-    /// it will print out the context.
-    #[cfg(feature = "std")]
-    #[cfg_attr(not(feature = "log_error_context"), allow(unused_variables))]
-    #[inline]
-    pub fn with_error_context<E: std::error::Error + ?Sized>(self, err: &E) -> Self {
```

Comment:
Praise: It's fine to get rid of this one since, as you said, we're only using the `Display` impl from the error.